src/app/qgscustomization.cpp

```diff
@@ -272,6 +272,14 @@ void QgsCustomizationDialog::apply()
 
 void QgsCustomizationDialog::cancel()
 {
+  if ( mSelectedWidgets.size() > 0 )
+  {
+    for ( int i = 0; i < mSelectedWidgets.size(); i++ )
+    {
+      mSelectedWidgets.at( i )->setStyleSheet( "" );
```

Comment:
This is a little risky (as is the existing code in `switchWidget`) as it potentially wipes out any existing stylesheet on the widget.

How about changing switchWidget to add something like:

     widget->setProperty( "originalStylesheet", widget->styleSheet() );

and then restoring that here?

    widget->setStyleSheet( widget->property( "originalStylesheet" ) );

Reply:
It seems that we should restore also in switchWidget